### PR TITLE
feat: :construction: add status and locked for project

### LINKED ISSUE #129: Statuts et verrouillage
# US
ETU lorsque j'effectue une opération sur un projet, un dépôt ou un utilisateur, je souhaite être informé de l'aboutissement de mon action.

## TODO
- ajouter clé `status: Enumeration` dans le schéma joi au niveau de `project`, `project.repos[n]`, `project.users[n]`, `project.envList[n]`
- pour `project.status` : `['initializing', 'created', 'failed', 'archived'] `
- pour `repos[n].status` - `users[n].status` - `envList[n].status` : `['initializing', 'created', 'failed', 'deleting'] `
- ajouter clé `locked: bool` dans le schéma joi au niveau de `project`

si `project.status || repos[n].status || users[n].status || env[n].status` sont différents de `created || failed` alors `locked: true`
sinon `locked: false`

### PR DESCRIPTION
**Pull request by clairenollet**
Closes #129

### CODE REVIEW

**Review by ArnaudTa — APPROVED** [review]

File: apps/client/src/components/RepoForm.vue [hunk 1]
```diff
@@ -15,6 +15,10 @@ const props = defineProps({
 })
 
 const localRepo = ref(props.repo)
+if (!localRepo.value.isInfra && !localRepo.value.isPrivate) {
+  localRepo.value.isInfra = false
+  localRepo.value.isPrivate = false
+}
```
[the remark this answers is not in the document]
Comment (clairenollet) [reply apps/client/src/components/RepoForm.vue:21]: `=== undefined`

File: apps/client/src/views/projects/DsoProjects.vue [hunk 2]
```diff
@@ -14,6 +14,7 @@ const setProjectList = (projects) => {
     id: project.id,
     title: project.projectName,
     to: `/projects/${project.id}/dashboard`,
+    status: project.status,
```
Comment (clairenollet) [inline apps/client/src/views/projects/DsoProjects.vue:17]: à suppr

File: apps/server/src/controllers/project.js [hunk 3]
```diff
@@ -42,24 +47,57 @@ export const createProjectController = async (req, res) => {
...
+
     send201(res, project)
   } catch (error) {
     req.log.error({
       ...getLogInfos(),
       description: 'Provisioning project with ansible failed',
       error,
     })
+    try {
+      project.status = 'failed'
```
Comment (clairenollet) [inline apps/server/src/controllers/project.js:86]: owner.status à failed

File: packages/shared/src/schemas/repo.js [hunk 4]
```diff
@@ -14,13 +15,18 @@ export const repoSchema = Joi.object({
     })
     .required(),
 
-  isPrivate: Joi.boolean(),
+  isPrivate: Joi.boolean()
+    .required(),
 
-  isInfra: Joi.boolean(),
+  isInfra: Joi.boolean()
+    .required(),
```
Comment (clairenollet) [inline packages/shared/src/schemas/repo.js:22]: commenter les `required()`

**Review by ArnaudTa — APPROVED** [review]

**Review by this-is-tobi — APPROVED** [review]

**Pull request merged**